lib/que/adapters/active_record_with_lock.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+module Que
+  module Adapters
+    class ActiveRecordWithLock < Que::Adapters::ActiveRecord
+      def initialize(job_connection_pool:, lock_connection_pool:)
+        @job_connection_pool = job_connection_pool
+        @lock_connection_pool = lock_connection_pool
+        super
+      end
+
+      def checkout
```

Comment:
Could we just override `checkout_activerecord_adapter` to checkout both connections? Looks like everything else in the method can be reused from the other adapter, simplifies the code a bit

Reply:
Added it here, in case we want to rescue any other Database related exceptions. But can move to checkout_activerecord_adapter

Reply:
updated